File: examples-sdk/go/002-basic-vilapp-gateway/main.go
// 002-basic-vilapp-gateway — Go SDK equivalent
// Compile: vil compile --from go --input 002-basic-vilapp-gateway/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("vil-app-gateway", 3081)

	gw := vil.NewService("gw")
	gw.Endpoint("POST", "/trigger", "trigger_handler")
	s.Service(gw)

	s.Compile()
}

File: examples-sdk/zig/002-basic-vilapp-gateway/main.zig
// 002-basic-vilapp-gateway — Zig SDK equivalent
// Compile: vil compile --from zig --input 002-basic-vilapp-gateway/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("vil-app-gateway", 3081);
    var gw = vil.Service.init("gw");
    gw.endpoint("POST", "/trigger", "trigger_handler");
    server.service(&gw);
    server.compile();
}
